### DISCUSSION TITLE
Design choice: reverse edges and vs undirected edges

### DISCUSSION THREAD

**Discussion by denadai2** [pyg-team/pytorch_geometric#7912] [Q&A] [answered]
Hi
Let's say I have an hetero network with `documentary` and `tvserie` in the graph. I have `documentary <has-same-author> tvserie`. Let's say I have

```
A <-> B
```
where A is 0 and B is 1.

 In `HeteroData` I could have 

- `edge_index = [[0, 1], [1, 0]]`
- `edge_index = [[0], [1]]` + rev_edges
- `edge_index = [[0, 1], [1, 0]]` + rev_edges

so I am confused about what's best and what's wrong. Could you help me on this?

thx

**Comment by rusty1s** [comment]
All edge indices in `HeteroData` objects needs to point to local node indices (i.e., they need to be in range `0` to `num_nodes - 1` of this particular node type). So if you have two node types with a single node each, the resulting `edge_index` should be defined as 
```python
data[tv, doc].edge_index = torch.tensor([[0], [0]])
data[doc, tv].edge_index = torch.tensor([[0], [0]])

**Comment by denadai2** [comment]
got it, sorry wrong example. However,

should I create rev_edges AND add flipped edges for undirected hetero graphs? (e.g. data[tv, doc].edge_index = torch.tensor([[0, 0], [0, 0]]))

**Reply by rusty1s** [reply, marked as answer]
No, one of them is enough. You can either create them manually or on the fly via `transforms.ToUndirected()`.

**Comment by denadai2** [comment]
sry I forgot to reply! Thannkkkss
